harness/determined/pytorch/_trainer.py

```diff
@@ -0,0 +1,211 @@
+import contextlib
+import logging
+import random
+from typing import Dict, Optional, Union, cast
+
+import torch
+import torch.distributed as dist
+
+import determined as det
+from determined import core, horovod, pytorch
+from determined.horovod import hvd
+from determined.profiler import DummyProfilerAgent, ProfilerAgent
+
+
+class Trainer:
+    def __init__(self, trial: pytorch.PyTorchTrial, context: pytorch.PyTorchTrialContext):
+        self._trial = trial
+        self._context = context
+        self._cluster_info = det.get_cluster_info()
+        self._core = self._context._core
+        self._distributed_backend = det._DistributedBackend()
+        self._det_profiler = DummyProfilerAgent()
+        self._trial_controller = None
+        self._local_training = self._cluster_info is None
+
+    def configure_profiler(
```

Comment:
are users allowed to call this multiple times?  Are they allowed to call it after training has already started?